src/main/java/eu/dissco/core/digitalspecimenprocessor/service/MidsService.java

```diff
@@ -54,88 +57,175 @@ private boolean compliesToMidsTwo(DigitalSpecimen attributes) {
+          attributes.getDwcRecordedById());
+    }
+  }
+
+  private boolean geographicalValuesComply(DigitalSpecimen attributes) {
+    if (attributes.getDwcOccurrence() != null
+        && attributes.getDwcOccurrence().get(0).getDctermsLocation() != null) {
+      var location = attributes.getDwcOccurrence().get(0).getDctermsLocation();
+      if (isValid(location.getDwcLocationID())) {
+        return true;
+      } else if (location.getGeoReference() != null) {
+        var georeference = location.getGeoReference();
+        return (georeference.getDwcDecimalLatitude() != null
+            && georeference.getDwcDecimalLongitude() != null
+            && isValid(georeference.getDwcGeodeticDatum())
+            && georeference.getDwcCoordinateUncertaintyInMeters() != null
+            && georeference.getDwcCoordinatePrecision() != null)
+            || (isValid(georeference.getDwcFootprintWkt())
+            && isValid(georeference.getDwcFootprintSrs()));
+      }
+    }
+    return false;
+  }
+
+  private boolean taxValuesComply(DigitalSpecimen attributes) {
+    var hasScientificNameId = false;
+    var hasIdentifiedById = false;
+    for (var identifications : attributes.getDwcIdentification()) {
+      if (isValid(identifications.getDwcIdentificationID())) {
+        hasIdentifiedById = true;
+      }
+      for (var taxonIdentification : identifications.getTaxonIdentifications()) {
+        if (isValid(taxonIdentification.getDwcScientificNameId())) {
+          hasScientificNameId = true;
+        }
+      }
+    }
+    return hasScientificNameId && hasIdentifiedById;
+  }
+
   private boolean compliesToMidsTwoBio(DigitalSpecimen attributes) {
-    return (attributes.getOdsMarkedAsType() != null && attributes.getOdsMarkedAsType())
+    return attributes.getOdsMarkedAsType() != null
         && (attributes.getOdsHasMedia() != null && attributes.getOdsHasMedia())
         && isQualitativeLocationValid(attributes)
         && isQuantitativeLocationValid(attributes)
-        && (occurrenceIsPresent(attributes) && isValid(
-        (attributes.getOccurrences().get(0).getDwcEventDate())))
-        && (occurrenceIsPresent(attributes) && isValid(
-        attributes.getOccurrences().get(0).getDwcFieldNumber()))
-        && isValid(attributes.getDwcRecordedBy());
+        && (occurrenceIsPresent(attributes) && hasValue(
+        attributes.getDwcOccurrence().get(0).getDwcEventDate(),
+        attributes.getDwcOccurrence().get(0).getDwcVerbatimEventDate(),
+        convertInteger(attributes.getDwcOccurrence().get(0).getDwcYear())))
+        && (occurrenceIsPresent(attributes) &&
+        hasValue(attributes.getDwcOccurrence().get(0).getDwcFieldNumber(),
+            attributes.getDwcOccurrence().get(0).getDwcRecordNumber()))
+        && hasValue(attributes.getDwcRecordedBy(), attributes.getDwcRecordedById());
+  }
+
+  private String convertInteger(Integer integer) {
```

Comment:
make static?

Reply:
It is only used here, if we use it in more place we can make a Utility class and put it in there